client/src/utils/jsonUtils.ts

```diff
@@ -16,11 +16,20 @@ export type JsonSchemaType = {
     | "array"
     | "object"
     | "null";
+  title?: string;
   description?: string;
-  required?: boolean;
+  required?: boolean | string[];
   default?: JsonValue;
   properties?: Record<string, JsonSchemaType>;
   items?: JsonSchemaType;
+  minimum?: number;
+  maximum?: number;
+  minLength?: number;
+  maxLength?: number;
+  pattern?: string;
+  format?: string;
+  enum?: string[];
+  enumNames?: string[];
```

Comment:
`enumNames` [is not part of the JSONSchema spec](https://github.com/rjsf-team/react-jsonschema-form/issues/532). The [recommended way to describe](https://github.com/json-schema-org/json-schema-spec/issues/57) enum labels is with `oneOf` or `anyOf`.